// ews-javascript-api/js/Enumerations/AutodiscoverErrorCode.js
"use strict";
(function (AutodiscoverErrorCode) {
    AutodiscoverErrorCode[AutodiscoverErrorCode["NoError"] = 0] = "NoError";
    AutodiscoverErrorCode[AutodiscoverErrorCode["RedirectAddress"] = 1] = "RedirectAddress";
    AutodiscoverErrorCode[AutodiscoverErrorCode["RedirectUrl"] = 2] = "RedirectUrl";
    AutodiscoverErrorCode[AutodiscoverErrorCode["InvalidUser"] = 3] = "InvalidUser";
    AutodiscoverErrorCode[AutodiscoverErrorCode["InvalidRequest"] = 4] = "InvalidRequest";
    AutodiscoverErrorCode[AutodiscoverErrorCode["InvalidSetting"] = 5] = "InvalidSetting";
    AutodiscoverErrorCode[AutodiscoverErrorCode["SettingIsNotAvailable"] = 6] = "SettingIsNotAvailable";
    AutodiscoverErrorCode[AutodiscoverErrorCode["ServerBusy"] = 7] = "ServerBusy";
    AutodiscoverErrorCode[AutodiscoverErrorCode["InvalidDomain"] = 8] = "InvalidDomain";
    AutodiscoverErrorCode[AutodiscoverErrorCode["NotFederated"] = 9] = "NotFederated";
    AutodiscoverErrorCode[AutodiscoverErrorCode["InternalServerError"] = 10] = "InternalServerError";
})(exports.AutodiscoverErrorCode || (exports.AutodiscoverErrorCode = {}));
var AutodiscoverErrorCode = exports.AutodiscoverErrorCode;
