// ews-javascript-api/js/Enumerations/DefaultExtendedPropertySet.js
"use strict";
(function (DefaultExtendedPropertySet) {
    DefaultExtendedPropertySet[DefaultExtendedPropertySet["Meeting"] = 0] = "Meeting";
    DefaultExtendedPropertySet[DefaultExtendedPropertySet["Appointment"] = 1] = "Appointment";
    DefaultExtendedPropertySet[DefaultExtendedPropertySet["Common"] = 2] = "Common";
    DefaultExtendedPropertySet[DefaultExtendedPropertySet["PublicStrings"] = 3] = "PublicStrings";
    DefaultExtendedPropertySet[DefaultExtendedPropertySet["Address"] = 4] = "Address";
    DefaultExtendedPropertySet[DefaultExtendedPropertySet["InternetHeaders"] = 5] = "InternetHeaders";
    DefaultExtendedPropertySet[DefaultExtendedPropertySet["CalendarAssistant"] = 6] = "CalendarAssistant";
    DefaultExtendedPropertySet[DefaultExtendedPropertySet["UnifiedMessaging"] = 7] = "UnifiedMessaging";
    DefaultExtendedPropertySet[DefaultExtendedPropertySet["Task"] = 8] = "Task";
})(exports.DefaultExtendedPropertySet || (exports.DefaultExtendedPropertySet = {}));
var DefaultExtendedPropertySet = exports.DefaultExtendedPropertySet;
