// ews-javascript-api/js/Enumerations/MapiPropertyType.js
"use strict";
(function (MapiPropertyType) {
    MapiPropertyType[MapiPropertyType["ApplicationTime"] = 0] = "ApplicationTime";
    MapiPropertyType[MapiPropertyType["ApplicationTimeArray"] = 1] = "ApplicationTimeArray";
    MapiPropertyType[MapiPropertyType["Binary"] = 2] = "Binary";
    MapiPropertyType[MapiPropertyType["BinaryArray"] = 3] = "BinaryArray";
    MapiPropertyType[MapiPropertyType["Boolean"] = 4] = "Boolean";
    MapiPropertyType[MapiPropertyType["CLSID"] = 5] = "CLSID";
    MapiPropertyType[MapiPropertyType["CLSIDArray"] = 6] = "CLSIDArray";
    MapiPropertyType[MapiPropertyType["Currency"] = 7] = "Currency";
    MapiPropertyType[MapiPropertyType["CurrencyArray"] = 8] = "CurrencyArray";
    MapiPropertyType[MapiPropertyType["Double"] = 9] = "Double";
    MapiPropertyType[MapiPropertyType["DoubleArray"] = 10] = "DoubleArray";
    MapiPropertyType[MapiPropertyType["Error"] = 11] = "Error";
    MapiPropertyType[MapiPropertyType["Float"] = 12] = "Float";
    MapiPropertyType[MapiPropertyType["FloatArray"] = 13] = "FloatArray";
    MapiPropertyType[MapiPropertyType["Integer"] = 14] = "Integer";
    MapiPropertyType[MapiPropertyType["IntegerArray"] = 15] = "IntegerArray";
    MapiPropertyType[MapiPropertyType["Long"] = 16] = "Long";
    MapiPropertyType[MapiPropertyType["LongArray"] = 17] = "LongArray";
    MapiPropertyType[MapiPropertyType["Null"] = 18] = "Null";
    MapiPropertyType[MapiPropertyType["Object"] = 19] = "Object";
    MapiPropertyType[MapiPropertyType["ObjectArray"] = 20] = "ObjectArray";
    MapiPropertyType[MapiPropertyType["Short"] = 21] = "Short";
    MapiPropertyType[MapiPropertyType["ShortArray"] = 22] = "ShortArray";
    MapiPropertyType[MapiPropertyType["SystemTime"] = 23] = "SystemTime";
    MapiPropertyType[MapiPropertyType["SystemTimeArray"] = 24] = "SystemTimeArray";
    MapiPropertyType[MapiPropertyType["String"] = 25] = "String";
    MapiPropertyType[MapiPropertyType["StringArray"] = 26] = "StringArray";
})(exports.MapiPropertyType || (exports.MapiPropertyType = {}));
var MapiPropertyType = exports.MapiPropertyType;
