// ews-javascript-api/js/Enumerations/PhoneNumberKey.js
"use strict";
(function (PhoneNumberKey) {
    PhoneNumberKey[PhoneNumberKey["AssistantPhone"] = 0] = "AssistantPhone";
    PhoneNumberKey[PhoneNumberKey["BusinessFax"] = 1] = "BusinessFax";
    PhoneNumberKey[PhoneNumberKey["BusinessPhone"] = 2] = "BusinessPhone";
    PhoneNumberKey[PhoneNumberKey["BusinessPhone2"] = 3] = "BusinessPhone2";
    PhoneNumberKey[PhoneNumberKey["Callback"] = 4] = "Callback";
    PhoneNumberKey[PhoneNumberKey["CarPhone"] = 5] = "CarPhone";
    PhoneNumberKey[PhoneNumberKey["CompanyMainPhone"] = 6] = "CompanyMainPhone";
    PhoneNumberKey[PhoneNumberKey["HomeFax"] = 7] = "HomeFax";
    PhoneNumberKey[PhoneNumberKey["HomePhone"] = 8] = "HomePhone";
    PhoneNumberKey[PhoneNumberKey["HomePhone2"] = 9] = "HomePhone2";
    PhoneNumberKey[PhoneNumberKey["Isdn"] = 10] = "Isdn";
    PhoneNumberKey[PhoneNumberKey["MobilePhone"] = 11] = "MobilePhone";
    PhoneNumberKey[PhoneNumberKey["OtherFax"] = 12] = "OtherFax";
    PhoneNumberKey[PhoneNumberKey["OtherTelephone"] = 13] = "OtherTelephone";
    PhoneNumberKey[PhoneNumberKey["Pager"] = 14] = "Pager";
    PhoneNumberKey[PhoneNumberKey["PrimaryPhone"] = 15] = "PrimaryPhone";
    PhoneNumberKey[PhoneNumberKey["RadioPhone"] = 16] = "RadioPhone";
    PhoneNumberKey[PhoneNumberKey["Telex"] = 17] = "Telex";
    PhoneNumberKey[PhoneNumberKey["TtyTddPhone"] = 18] = "TtyTddPhone";
})(exports.PhoneNumberKey || (exports.PhoneNumberKey = {}));
var PhoneNumberKey = exports.PhoneNumberKey;
