// ews-javascript-api/js/Enumerations/RuleProperty.js
"use strict";
(function (RuleProperty) {
    RuleProperty[RuleProperty["RuleId"] = 0] = "RuleId";
    RuleProperty[RuleProperty["DisplayName"] = 1] = "DisplayName";
    RuleProperty[RuleProperty["Priority"] = 2] = "Priority";
    RuleProperty[RuleProperty["IsNotSupported"] = 3] = "IsNotSupported";
    RuleProperty[RuleProperty["Actions"] = 4] = "Actions";
    RuleProperty[RuleProperty["ConditionCategories"] = 5] = "ConditionCategories";
    RuleProperty[RuleProperty["ConditionContainsBodyStrings"] = 6] = "ConditionContainsBodyStrings";
    RuleProperty[RuleProperty["ConditionContainsHeaderStrings"] = 7] = "ConditionContainsHeaderStrings";
    RuleProperty[RuleProperty["ConditionContainsRecipientStrings"] = 8] = "ConditionContainsRecipientStrings";
    RuleProperty[RuleProperty["ConditionContainsSenderStrings"] = 9] = "ConditionContainsSenderStrings";
    RuleProperty[RuleProperty["ConditionContainsSubjectOrBodyStrings"] = 10] = "ConditionContainsSubjectOrBodyStrings";
    RuleProperty[RuleProperty["ConditionContainsSubjectStrings"] = 11] = "ConditionContainsSubjectStrings";
    RuleProperty[RuleProperty["ConditionFlaggedForAction"] = 12] = "ConditionFlaggedForAction";
    RuleProperty[RuleProperty["ConditionFromAddresses"] = 13] = "ConditionFromAddresses";
    RuleProperty[RuleProperty["ConditionFromConnectedAccounts"] = 14] = "ConditionFromConnectedAccounts";
    RuleProperty[RuleProperty["ConditionHasAttachments"] = 15] = "ConditionHasAttachments";
    RuleProperty[RuleProperty["ConditionImportance"] = 16] = "ConditionImportance";
    RuleProperty[RuleProperty["ConditionIsApprovalRequest"] = 17] = "ConditionIsApprovalRequest";
    RuleProperty[RuleProperty["ConditionIsAutomaticForward"] = 18] = "ConditionIsAutomaticForward";
    RuleProperty[RuleProperty["ConditionIsAutomaticReply"] = 19] = "ConditionIsAutomaticReply";
    RuleProperty[RuleProperty["ConditionIsEncrypted"] = 20] = "ConditionIsEncrypted";
    RuleProperty[RuleProperty["ConditionIsMeetingRequest"] = 21] = "ConditionIsMeetingRequest";
    RuleProperty[RuleProperty["ConditionIsMeetingResponse"] = 22] = "ConditionIsMeetingResponse";
    RuleProperty[RuleProperty["ConditionIsNonDeliveryReport"] = 23] = "ConditionIsNonDeliveryReport";
    RuleProperty[RuleProperty["ConditionIsPermissionControlled"] = 24] = "ConditionIsPermissionControlled";
    RuleProperty[RuleProperty["ConditionIsRead"] = 25] = "ConditionIsRead";
    RuleProperty[RuleProperty["ConditionIsSigned"] = 26] = "ConditionIsSigned";
    RuleProperty[RuleProperty["ConditionIsVoicemail"] = 27] = "ConditionIsVoicemail";
    RuleProperty[RuleProperty["ConditionIsReadReceipt"] = 28] = "ConditionIsReadReceipt";
    RuleProperty[RuleProperty["ConditionItemClasses"] = 29] = "ConditionItemClasses";
    RuleProperty[RuleProperty["ConditionMessageClassifications"] = 30] = "ConditionMessageClassifications";
    RuleProperty[RuleProperty["ConditionNotSentToMe"] = 31] = "ConditionNotSentToMe";
    RuleProperty[RuleProperty["ConditionSentCcMe"] = 32] = "ConditionSentCcMe";
    RuleProperty[RuleProperty["ConditionSentOnlyToMe"] = 33] = "ConditionSentOnlyToMe";
    RuleProperty[RuleProperty["ConditionSentToAddresses"] = 34] = "ConditionSentToAddresses";
    RuleProperty[RuleProperty["ConditionSentToMe"] = 35] = "ConditionSentToMe";
    RuleProperty[RuleProperty["ConditionSentToOrCcMe"] = 36] = "ConditionSentToOrCcMe";
    RuleProperty[RuleProperty["ConditionSensitivity"] = 37] = "ConditionSensitivity";
    RuleProperty[RuleProperty["ConditionWithinDateRange"] = 38] = "ConditionWithinDateRange";
    RuleProperty[RuleProperty["ConditionWithinSizeRange"] = 39] = "ConditionWithinSizeRange";
    RuleProperty[RuleProperty["ExceptionCategories"] = 40] = "ExceptionCategories";
    RuleProperty[RuleProperty["ExceptionContainsBodyStrings"] = 41] = "ExceptionContainsBodyStrings";
    RuleProperty[RuleProperty["ExceptionContainsHeaderStrings"] = 42] = "ExceptionContainsHeaderStrings";
    RuleProperty[RuleProperty["ExceptionContainsRecipientStrings"] = 43] = "ExceptionContainsRecipientStrings";
    RuleProperty[RuleProperty["ExceptionContainsSenderStrings"] = 44] = "ExceptionContainsSenderStrings";
    RuleProperty[RuleProperty["ExceptionContainsSubjectOrBodyStrings"] = 45] = "ExceptionContainsSubjectOrBodyStrings";
    RuleProperty[RuleProperty["ExceptionContainsSubjectStrings"] = 46] = "ExceptionContainsSubjectStrings";
    RuleProperty[RuleProperty["ExceptionFlaggedForAction"] = 47] = "ExceptionFlaggedForAction";
    RuleProperty[RuleProperty["ExceptionFromAddresses"] = 48] = "ExceptionFromAddresses";
    RuleProperty[RuleProperty["ExceptionFromConnectedAccounts"] = 49] = "ExceptionFromConnectedAccounts";
    RuleProperty[RuleProperty["ExceptionHasAttachments"] = 50] = "ExceptionHasAttachments";
    RuleProperty[RuleProperty["ExceptionImportance"] = 51] = "ExceptionImportance";
    RuleProperty[RuleProperty["ExceptionIsApprovalRequest"] = 52] = "ExceptionIsApprovalRequest";
    RuleProperty[RuleProperty["ExceptionIsAutomaticForward"] = 53] = "ExceptionIsAutomaticForward";
    RuleProperty[RuleProperty["ExceptionIsAutomaticReply"] = 54] = "ExceptionIsAutomaticReply";
    RuleProperty[RuleProperty["ExceptionIsEncrypted"] = 55] = "ExceptionIsEncrypted";
    RuleProperty[RuleProperty["ExceptionIsMeetingRequest"] = 56] = "ExceptionIsMeetingRequest";
    RuleProperty[RuleProperty["ExceptionIsMeetingResponse"] = 57] = "ExceptionIsMeetingResponse";
    RuleProperty[RuleProperty["ExceptionIsNonDeliveryReport"] = 58] = "ExceptionIsNonDeliveryReport";
    RuleProperty[RuleProperty["ExceptionIsPermissionControlled"] = 59] = "ExceptionIsPermissionControlled";
    RuleProperty[RuleProperty["ExceptionIsRead"] = 60] = "ExceptionIsRead";
    RuleProperty[RuleProperty["ExceptionIsSigned"] = 61] = "ExceptionIsSigned";
    RuleProperty[RuleProperty["ExceptionIsVoicemail"] = 62] = "ExceptionIsVoicemail";
    RuleProperty[RuleProperty["ExceptionItemClasses"] = 63] = "ExceptionItemClasses";
    RuleProperty[RuleProperty["ExceptionMessageClassifications"] = 64] = "ExceptionMessageClassifications";
    RuleProperty[RuleProperty["ExceptionNotSentToMe"] = 65] = "ExceptionNotSentToMe";
    RuleProperty[RuleProperty["ExceptionSentCcMe"] = 66] = "ExceptionSentCcMe";
    RuleProperty[RuleProperty["ExceptionSentOnlyToMe"] = 67] = "ExceptionSentOnlyToMe";
    RuleProperty[RuleProperty["ExceptionSentToAddresses"] = 68] = "ExceptionSentToAddresses";
    RuleProperty[RuleProperty["ExceptionSentToMe"] = 69] = "ExceptionSentToMe";
    RuleProperty[RuleProperty["ExceptionSentToOrCcMe"] = 70] = "ExceptionSentToOrCcMe";
    RuleProperty[RuleProperty["ExceptionSensitivity"] = 71] = "ExceptionSensitivity";
    RuleProperty[RuleProperty["ExceptionWithinDateRange"] = 72] = "ExceptionWithinDateRange";
    RuleProperty[RuleProperty["ExceptionWithinSizeRange"] = 73] = "ExceptionWithinSizeRange";
    RuleProperty[RuleProperty["ActionCategories"] = 74] = "ActionCategories";
    RuleProperty[RuleProperty["ActionCopyToFolder"] = 75] = "ActionCopyToFolder";
    RuleProperty[RuleProperty["ActionDelete"] = 76] = "ActionDelete";
    RuleProperty[RuleProperty["ActionForwardAsAttachmentToRecipients"] = 77] = "ActionForwardAsAttachmentToRecipients";
    RuleProperty[RuleProperty["ActionForwardToRecipients"] = 78] = "ActionForwardToRecipients";
    RuleProperty[RuleProperty["ActionImportance"] = 79] = "ActionImportance";
    RuleProperty[RuleProperty["ActionMarkAsRead"] = 80] = "ActionMarkAsRead";
    RuleProperty[RuleProperty["ActionMoveToFolder"] = 81] = "ActionMoveToFolder";
    RuleProperty[RuleProperty["ActionPermanentDelete"] = 82] = "ActionPermanentDelete";
    RuleProperty[RuleProperty["ActionRedirectToRecipients"] = 83] = "ActionRedirectToRecipients";
    RuleProperty[RuleProperty["ActionSendSMSAlertToRecipients"] = 84] = "ActionSendSMSAlertToRecipients";
    RuleProperty[RuleProperty["ActionServerReplyWithMessage"] = 85] = "ActionServerReplyWithMessage";
    RuleProperty[RuleProperty["ActionStopProcessingRules"] = 86] = "ActionStopProcessingRules";
    RuleProperty[RuleProperty["IsEnabled"] = 87] = "IsEnabled";
    RuleProperty[RuleProperty["IsInError"] = 88] = "IsInError";
    RuleProperty[RuleProperty["Conditions"] = 89] = "Conditions";
    RuleProperty[RuleProperty["Exceptions"] = 90] = "Exceptions";
})(exports.RuleProperty || (exports.RuleProperty = {}));
var RuleProperty = exports.RuleProperty;
