// ews-javascript-api/js/Enumerations/UserSettingName.js
"use strict";
(function (UserSettingName) {
    UserSettingName[UserSettingName["UserDisplayName"] = 0] = "UserDisplayName";
    UserSettingName[UserSettingName["UserDN"] = 1] = "UserDN";
    UserSettingName[UserSettingName["UserDeploymentId"] = 2] = "UserDeploymentId";
    UserSettingName[UserSettingName["InternalMailboxServer"] = 3] = "InternalMailboxServer";
    UserSettingName[UserSettingName["InternalRpcClientServer"] = 4] = "InternalRpcClientServer";
    UserSettingName[UserSettingName["InternalMailboxServerDN"] = 5] = "InternalMailboxServerDN";
    UserSettingName[UserSettingName["InternalEcpUrl"] = 6] = "InternalEcpUrl";
    UserSettingName[UserSettingName["InternalEcpVoicemailUrl"] = 7] = "InternalEcpVoicemailUrl";
    UserSettingName[UserSettingName["InternalEcpEmailSubscriptionsUrl"] = 8] = "InternalEcpEmailSubscriptionsUrl";
    UserSettingName[UserSettingName["InternalEcpTextMessagingUrl"] = 9] = "InternalEcpTextMessagingUrl";
    UserSettingName[UserSettingName["InternalEcpDeliveryReportUrl"] = 10] = "InternalEcpDeliveryReportUrl";
    UserSettingName[UserSettingName["InternalEcpRetentionPolicyTagsUrl"] = 11] = "InternalEcpRetentionPolicyTagsUrl";
    UserSettingName[UserSettingName["InternalEcpPublishingUrl"] = 12] = "InternalEcpPublishingUrl";
    UserSettingName[UserSettingName["InternalEcpPhotoUrl"] = 13] = "InternalEcpPhotoUrl";
    UserSettingName[UserSettingName["InternalEcpConnectUrl"] = 14] = "InternalEcpConnectUrl";
    UserSettingName[UserSettingName["InternalEcpTeamMailboxUrl"] = 15] = "InternalEcpTeamMailboxUrl";
    UserSettingName[UserSettingName["InternalEcpTeamMailboxCreatingUrl"] = 16] = "InternalEcpTeamMailboxCreatingUrl";
    UserSettingName[UserSettingName["InternalEcpTeamMailboxEditingUrl"] = 17] = "InternalEcpTeamMailboxEditingUrl";
    UserSettingName[UserSettingName["InternalEcpTeamMailboxHidingUrl"] = 18] = "InternalEcpTeamMailboxHidingUrl";
    UserSettingName[UserSettingName["InternalEcpExtensionInstallationUrl"] = 19] = "InternalEcpExtensionInstallationUrl";
    UserSettingName[UserSettingName["InternalEwsUrl"] = 20] = "InternalEwsUrl";
    UserSettingName[UserSettingName["InternalEmwsUrl"] = 21] = "InternalEmwsUrl";
    UserSettingName[UserSettingName["InternalOABUrl"] = 22] = "InternalOABUrl";
    UserSettingName[UserSettingName["InternalPhotosUrl"] = 23] = "InternalPhotosUrl";
    UserSettingName[UserSettingName["InternalUMUrl"] = 24] = "InternalUMUrl";
    UserSettingName[UserSettingName["InternalWebClientUrls"] = 25] = "InternalWebClientUrls";
    UserSettingName[UserSettingName["MailboxDN"] = 26] = "MailboxDN";
    UserSettingName[UserSettingName["PublicFolderServer"] = 27] = "PublicFolderServer";
    UserSettingName[UserSettingName["ActiveDirectoryServer"] = 28] = "ActiveDirectoryServer";
    UserSettingName[UserSettingName["ExternalMailboxServer"] = 29] = "ExternalMailboxServer";
    UserSettingName[UserSettingName["ExternalMailboxServerRequiresSSL"] = 30] = "ExternalMailboxServerRequiresSSL";
    UserSettingName[UserSettingName["ExternalMailboxServerAuthenticationMethods"] = 31] = "ExternalMailboxServerAuthenticationMethods";
    UserSettingName[UserSettingName["EcpVoicemailUrlFragment"] = 32] = "EcpVoicemailUrlFragment";
    UserSettingName[UserSettingName["EcpEmailSubscriptionsUrlFragment"] = 33] = "EcpEmailSubscriptionsUrlFragment";
    UserSettingName[UserSettingName["EcpTextMessagingUrlFragment"] = 34] = "EcpTextMessagingUrlFragment";
    UserSettingName[UserSettingName["EcpDeliveryReportUrlFragment"] = 35] = "EcpDeliveryReportUrlFragment";
    UserSettingName[UserSettingName["EcpRetentionPolicyTagsUrlFragment"] = 36] = "EcpRetentionPolicyTagsUrlFragment";
    UserSettingName[UserSettingName["EcpPublishingUrlFragment"] = 37] = "EcpPublishingUrlFragment";
    UserSettingName[UserSettingName["EcpPhotoUrlFragment"] = 38] = "EcpPhotoUrlFragment";
    UserSettingName[UserSettingName["EcpConnectUrlFragment"] = 39] = "EcpConnectUrlFragment";
    UserSettingName[UserSettingName["EcpTeamMailboxUrlFragment"] = 40] = "EcpTeamMailboxUrlFragment";
    UserSettingName[UserSettingName["EcpTeamMailboxCreatingUrlFragment"] = 41] = "EcpTeamMailboxCreatingUrlFragment";
    UserSettingName[UserSettingName["EcpTeamMailboxEditingUrlFragment"] = 42] = "EcpTeamMailboxEditingUrlFragment";
    UserSettingName[UserSettingName["EcpExtensionInstallationUrlFragment"] = 43] = "EcpExtensionInstallationUrlFragment";
    UserSettingName[UserSettingName["ExternalEcpUrl"] = 44] = "ExternalEcpUrl";
    UserSettingName[UserSettingName["ExternalEcpVoicemailUrl"] = 45] = "ExternalEcpVoicemailUrl";
    UserSettingName[UserSettingName["ExternalEcpEmailSubscriptionsUrl"] = 46] = "ExternalEcpEmailSubscriptionsUrl";
    UserSettingName[UserSettingName["ExternalEcpTextMessagingUrl"] = 47] = "ExternalEcpTextMessagingUrl";
    UserSettingName[UserSettingName["ExternalEcpDeliveryReportUrl"] = 48] = "ExternalEcpDeliveryReportUrl";
    UserSettingName[UserSettingName["ExternalEcpRetentionPolicyTagsUrl"] = 49] = "ExternalEcpRetentionPolicyTagsUrl";
    UserSettingName[UserSettingName["ExternalEcpPublishingUrl"] = 50] = "ExternalEcpPublishingUrl";
    UserSettingName[UserSettingName["ExternalEcpPhotoUrl"] = 51] = "ExternalEcpPhotoUrl";
    UserSettingName[UserSettingName["ExternalEcpConnectUrl"] = 52] = "ExternalEcpConnectUrl";
    UserSettingName[UserSettingName["ExternalEcpTeamMailboxUrl"] = 53] = "ExternalEcpTeamMailboxUrl";
    UserSettingName[UserSettingName["ExternalEcpTeamMailboxCreatingUrl"] = 54] = "ExternalEcpTeamMailboxCreatingUrl";
    UserSettingName[UserSettingName["ExternalEcpTeamMailboxEditingUrl"] = 55] = "ExternalEcpTeamMailboxEditingUrl";
    UserSettingName[UserSettingName["ExternalEcpTeamMailboxHidingUrl"] = 56] = "ExternalEcpTeamMailboxHidingUrl";
    UserSettingName[UserSettingName["ExternalEcpExtensionInstallationUrl"] = 57] = "ExternalEcpExtensionInstallationUrl";
    UserSettingName[UserSettingName["ExternalEwsUrl"] = 58] = "ExternalEwsUrl";
    UserSettingName[UserSettingName["ExternalEmwsUrl"] = 59] = "ExternalEmwsUrl";
    UserSettingName[UserSettingName["ExternalOABUrl"] = 60] = "ExternalOABUrl";
    UserSettingName[UserSettingName["ExternalPhotosUrl"] = 61] = "ExternalPhotosUrl";
    UserSettingName[UserSettingName["ExternalUMUrl"] = 62] = "ExternalUMUrl";
    UserSettingName[UserSettingName["ExternalWebClientUrls"] = 63] = "ExternalWebClientUrls";
    UserSettingName[UserSettingName["CrossOrganizationSharingEnabled"] = 64] = "CrossOrganizationSharingEnabled";
    UserSettingName[UserSettingName["AlternateMailboxes"] = 65] = "AlternateMailboxes";
    UserSettingName[UserSettingName["CasVersion"] = 66] = "CasVersion";
    UserSettingName[UserSettingName["EwsSupportedSchemas"] = 67] = "EwsSupportedSchemas";
    UserSettingName[UserSettingName["InternalPop3Connections"] = 68] = "InternalPop3Connections";
    UserSettingName[UserSettingName["ExternalPop3Connections"] = 69] = "ExternalPop3Connections";
    UserSettingName[UserSettingName["InternalImap4Connections"] = 70] = "InternalImap4Connections";
    UserSettingName[UserSettingName["ExternalImap4Connections"] = 71] = "ExternalImap4Connections";
    UserSettingName[UserSettingName["InternalSmtpConnections"] = 72] = "InternalSmtpConnections";
    UserSettingName[UserSettingName["ExternalSmtpConnections"] = 73] = "ExternalSmtpConnections";
    UserSettingName[UserSettingName["InternalServerExclusiveConnect"] = 74] = "InternalServerExclusiveConnect";
    UserSettingName[UserSettingName["ExternalEwsVersion"] = 75] = "ExternalEwsVersion";
    UserSettingName[UserSettingName["MobileMailboxPolicy"] = 76] = "MobileMailboxPolicy";
    UserSettingName[UserSettingName["DocumentSharingLocations"] = 77] = "DocumentSharingLocations";
    UserSettingName[UserSettingName["UserMSOnline"] = 78] = "UserMSOnline";
    UserSettingName[UserSettingName["InternalMailboxServerAuthenticationMethods"] = 79] = "InternalMailboxServerAuthenticationMethods";
    UserSettingName[UserSettingName["MailboxVersion"] = 80] = "MailboxVersion";
    UserSettingName[UserSettingName["SPMySiteHostURL"] = 81] = "SPMySiteHostURL";
    UserSettingName[UserSettingName["SiteMailboxCreationURL"] = 82] = "SiteMailboxCreationURL";
    UserSettingName[UserSettingName["InternalRpcHttpServer"] = 83] = "InternalRpcHttpServer";
    UserSettingName[UserSettingName["InternalRpcHttpConnectivityRequiresSsl"] = 84] = "InternalRpcHttpConnectivityRequiresSsl";
    UserSettingName[UserSettingName["InternalRpcHttpAuthenticationMethod"] = 85] = "InternalRpcHttpAuthenticationMethod";
    UserSettingName[UserSettingName["ExternalServerExclusiveConnect"] = 86] = "ExternalServerExclusiveConnect";
    UserSettingName[UserSettingName["ExchangeRpcUrl"] = 87] = "ExchangeRpcUrl";
    UserSettingName[UserSettingName["ShowGalAsDefaultView"] = 88] = "ShowGalAsDefaultView";
    UserSettingName[UserSettingName["AutoDiscoverSMTPAddress"] = 89] = "AutoDiscoverSMTPAddress";
    UserSettingName[UserSettingName["InteropExternalEwsUrl"] = 90] = "InteropExternalEwsUrl";
    UserSettingName[UserSettingName["InteropExternalEwsVersion"] = 91] = "InteropExternalEwsVersion";
    UserSettingName[UserSettingName["PublicFolderInformation"] = 92] = "PublicFolderInformation";
    UserSettingName[UserSettingName["RedirectUrl"] = 93] = "RedirectUrl";
    UserSettingName[UserSettingName["EwsPartnerUrl"] = 94] = "EwsPartnerUrl";
    UserSettingName[UserSettingName["CertPrincipalName"] = 95] = "CertPrincipalName";
    UserSettingName[UserSettingName["GroupingInformation"] = 96] = "GroupingInformation";
    UserSettingName[UserSettingName["MapiHttpEnabled"] = 97] = "MapiHttpEnabled";
})(exports.UserSettingName || (exports.UserSettingName = {}));
var UserSettingName = exports.UserSettingName;
