// ews-javascript-api/js/Enumerations/UserConfigurationDictionaryObjectType.js
"use strict";
(function (UserConfigurationDictionaryObjectType) {
    /**
     * DateTime type.
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["DateTime"] = 0] = "DateTime";
    /**
     * Boolean type.
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["Boolean"] = 1] = "Boolean";
    /**
     * Byte type.
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["Byte"] = 2] = "Byte";
    /**
     * String type.
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["String"] = 3] = "String";
    /**
     * 32-bit integer type.
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["Integer32"] = 4] = "Integer32";
    /**
     * 32-bit unsigned integer type.
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["UnsignedInteger32"] = 5] = "UnsignedInteger32";
    /**
     * 64-bit integer type.
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["Integer64"] = 6] = "Integer64";
    /**
     * 64-bit unsigned integer type.
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["UnsignedInteger64"] = 7] = "UnsignedInteger64";
    /**
     * String array type.
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["StringArray"] = 8] = "StringArray";
    /**
     * Byte array type
     */
    UserConfigurationDictionaryObjectType[UserConfigurationDictionaryObjectType["ByteArray"] = 9] = "ByteArray";
})(exports.UserConfigurationDictionaryObjectType || (exports.UserConfigurationDictionaryObjectType = {}));
var UserConfigurationDictionaryObjectType = exports.UserConfigurationDictionaryObjectType;
