// ews-javascript-api/js/Enumerations/WellKnownFolderName.js
"use strict";
(function (WellKnownFolderName) {
    WellKnownFolderName[WellKnownFolderName["Calendar"] = 0] = "Calendar";
    WellKnownFolderName[WellKnownFolderName["Contacts"] = 1] = "Contacts";
    WellKnownFolderName[WellKnownFolderName["DeletedItems"] = 2] = "DeletedItems";
    WellKnownFolderName[WellKnownFolderName["Drafts"] = 3] = "Drafts";
    WellKnownFolderName[WellKnownFolderName["Inbox"] = 4] = "Inbox";
    WellKnownFolderName[WellKnownFolderName["Journal"] = 5] = "Journal";
    WellKnownFolderName[WellKnownFolderName["Notes"] = 6] = "Notes";
    WellKnownFolderName[WellKnownFolderName["Outbox"] = 7] = "Outbox";
    WellKnownFolderName[WellKnownFolderName["SentItems"] = 8] = "SentItems";
    WellKnownFolderName[WellKnownFolderName["Tasks"] = 9] = "Tasks";
    WellKnownFolderName[WellKnownFolderName["MsgFolderRoot"] = 10] = "MsgFolderRoot";
    WellKnownFolderName[WellKnownFolderName["PublicFoldersRoot"] = 11] = "PublicFoldersRoot";
    WellKnownFolderName[WellKnownFolderName["Root"] = 12] = "Root";
    WellKnownFolderName[WellKnownFolderName["JunkEmail"] = 13] = "JunkEmail";
    WellKnownFolderName[WellKnownFolderName["SearchFolders"] = 14] = "SearchFolders";
    WellKnownFolderName[WellKnownFolderName["VoiceMail"] = 15] = "VoiceMail";
    WellKnownFolderName[WellKnownFolderName["RecoverableItemsRoot"] = 16] = "RecoverableItemsRoot";
    WellKnownFolderName[WellKnownFolderName["RecoverableItemsDeletions"] = 17] = "RecoverableItemsDeletions";
    WellKnownFolderName[WellKnownFolderName["RecoverableItemsVersions"] = 18] = "RecoverableItemsVersions";
    WellKnownFolderName[WellKnownFolderName["RecoverableItemsPurges"] = 19] = "RecoverableItemsPurges";
    WellKnownFolderName[WellKnownFolderName["ArchiveRoot"] = 20] = "ArchiveRoot";
    WellKnownFolderName[WellKnownFolderName["ArchiveMsgFolderRoot"] = 21] = "ArchiveMsgFolderRoot";
    WellKnownFolderName[WellKnownFolderName["ArchiveDeletedItems"] = 22] = "ArchiveDeletedItems";
    WellKnownFolderName[WellKnownFolderName["ArchiveRecoverableItemsRoot"] = 23] = "ArchiveRecoverableItemsRoot";
    WellKnownFolderName[WellKnownFolderName["ArchiveRecoverableItemsDeletions"] = 24] = "ArchiveRecoverableItemsDeletions";
    WellKnownFolderName[WellKnownFolderName["ArchiveRecoverableItemsVersions"] = 25] = "ArchiveRecoverableItemsVersions";
    WellKnownFolderName[WellKnownFolderName["ArchiveRecoverableItemsPurges"] = 26] = "ArchiveRecoverableItemsPurges";
    WellKnownFolderName[WellKnownFolderName["SyncIssues"] = 27] = "SyncIssues";
    WellKnownFolderName[WellKnownFolderName["Conflicts"] = 28] = "Conflicts";
    WellKnownFolderName[WellKnownFolderName["LocalFailures"] = 29] = "LocalFailures";
    WellKnownFolderName[WellKnownFolderName["ServerFailures"] = 30] = "ServerFailures";
    WellKnownFolderName[WellKnownFolderName["RecipientCache"] = 31] = "RecipientCache";
    WellKnownFolderName[WellKnownFolderName["QuickContacts"] = 32] = "QuickContacts";
    WellKnownFolderName[WellKnownFolderName["ConversationHistory"] = 33] = "ConversationHistory";
    WellKnownFolderName[WellKnownFolderName["ToDoSearch"] = 34] = "ToDoSearch";
})(exports.WellKnownFolderName || (exports.WellKnownFolderName = {}));
var WellKnownFolderName = exports.WellKnownFolderName;
